inference-engine/thirdparty/clDNN/tests/test_cases/reorder_gpu_test.cpp

```diff
@@ -23,8 +23,249 @@ using namespace cldnn;
 using namespace tests;
 using namespace testing;
 
-TEST(reorder_gpu_f32, basic)
-{
+static void compare_bfyx2blocked_with_ref(const std::string& kernel_name,
+    const data_types input_data_type, const data_types output_data_type,
+    cldnn::format input_format, cldnn::format output_format,
+    int32_t b_in, int32_t f_in, int32_t x_in, int32_t y_in, int32_t z_in = 0, int32_t w_in = 0) {
+    const auto& engine = get_test_engine();
+
+    tensor ts;
+    if (input_format.dimension() == 4) {
+        ts = { b_in, f_in, x_in, y_in };
+    }
+    else if (input_format.dimension() == 5) {
+        ts = { b_in, f_in, x_in, y_in, z_in };
+    }
+    else {
+        ts = { b_in, f_in, x_in, y_in, z_in, w_in };
+    }
+
+    auto input = memory::allocate(engine, { input_data_type, input_format, ts });
+    layout output_layout(output_data_type, output_format, ts);
+
+    if (input_data_type == data_types::i8) {
+        auto input_ptr = input.pointer<unsigned char>();
+        unsigned char i = 1;
+        for (auto it = input_ptr.begin(); it != input_ptr.end(); ++it)
+        {
+            *it = (i++);
+            if (i > 100) {
```

Comment:
Now we have generate_random_2d, 3d, 4d, ... methods, but as for me new "generate_counter" method is really needed. Maybe it makes sence to add these functions to "test_utils.h" to simplify the same code? Also it's a very useful feature especially for int8 debug.

Reply:
Maybe something looks like:
```
template<typename T>
std::vector<std::vector<T>> generate_counter_2d(size_t a, size_t b, size_t begin) {
    std::vector<std::vector<T>> v(a);
    for (size_t i = 0; i < a; ++i)
        v[i] = generate_counter_1d<T>(b, begin + i * b);
    return v;
}

template<typename T>
std::vector<std::vector<std::vector<T>>> generate_counter_3d(size_t a, size_t b, size_t c, size_t begin) {
    std::vector<std::vector<std::vector<T>>> v(a);
    for (size_t i = 0; i < a; ++i)
        v[i] = generate_counter_2d<T>(b, c, begin + i * b * c);
    return v;
}

template<typename T>
std::vector<std::vector<std::vector<std::vector<T>>>> generate_counter_4d(size_t a, size_t b, size_t c, size_t d, size_t begin) {
    std::vector<std::vector<std::vector<std::vector<T>>>> v(a);
    for (size_t i = 0; i < a; ++i)
        v[i] = generate_counter_3d<T>(b, c, d, begin + i * b * c * d);
    return v;
}
```

Reply:
Basically, I agree with your suggestion to add "generate_counter" because I also feel current generate_random is not enough. However, this is a kind of new feature in terms of optimizing 'reorder' so that I'll add this utility function on future PR which is not urgent.